packages/devtools_app/lib/src/charts/flame_chart.dart

```diff
@@ -25,9 +25,16 @@ import '../ui/search.dart';
 import '../utils.dart';
 
 const double rowPadding = 2.0;
-const double rowHeight = 25.0;
-const double rowHeightWithPadding = rowHeight + rowPadding;
-const double sectionSpacing = 16.0;
+// Flame chart rows contain text so are not readable if they do not scale with
+// the font factor.
+double get rowHeight => scaleByFontFactor(25.0);
+double get rowHeightWithPadding => rowHeight + rowPadding;
+
+// This spacing needs to be scaled by the font factor otherwise section
+// labels will not have enough room. Typically spacing values should not depend
+// on the font size scale factor. TODO(jacobr): clean up the section spacing so
+// it is not used in a case where it is not really spacing.
+double get sectionSpacing => scaleByFontFactor(16.0);
```

Comment:
@kenzieschmoll any idea about this? the flame chart starts throwing negative padding value exceptions if you leave this as 16.0

Reply:
negative padding in the X or Y direction?

Reply:
-23.0 top padding (Y). topSpacer is computed as -23
I don't quite understand why rowOffsetForTopPadding is multiplied by sectionSpacing.

Reply:
This is surprising to me that this needs to be scaled at all. This is supposed to be the vertical spacing between sections of the flame chart, and should be independent of the size of the rows (which would be affected by font size).

`rowOffsetForTopPadding` is the number of rows we want to offset for padding at the top of the flame chart. For example, we need more rows of padding in the timeline flame chart that has the timestamp grid at the top than we need in the Cpu profiler flame chart, which doesn't have anything to account for at the top.

Reply:
Filed https://github.com/flutter/devtools/issues/3306 to track. I'm not sure how to fix this.